src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,38 +52,53 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	static String checkRpid(final String putativeRpid, final String name) {
+		final String rpid = requireNonNull(processString(putativeRpid), name);
+
+		final Pattern validRpidRegex = Pattern.compile("^\\w+:\\w+");
```

Comment:
I deliberately left the regex open-ended (no `$`) to allow for any of a myriad of different dbxrefs that I've encountered on my journeys. DOIs can contain all manner of non-alphanumeric characters, so it wouldn't be just accounting for `/` and `.`, and this field also has to accommodate references from other databases that may also play fast and loose with sane standards for PID naming.

This ties in somewhat to the comments below about enums and where validation should occur.

Reply:
This is true for the prefix portion of the ID as well?

The reason I was a bit confused is that if there's no anchor, there's no difference between `\w` and `\w+` - both enforce one alphnum character and then there's no constraints on anything after that. The `+` made me think there was a missing anchor.

I'm also wondering, if dbxrefs are so unconstrained, if there should be a constraint on the first character?

Reply:
I've changed the regex so that it now just ensures that the first character after the colon is not whitespace. The potential prefixes and prefix formatting should be constrained by the Credit Engine, and the CE should also ensure that the whole thing is in a reasonable form.

Reply:
that's fine, but if there's some general restrictions that are relatively easy to do I don't think there's a reason not to handle them here as well just to be safe

Reply:
The other restrictions are going to be checked by an external python lib, so not really suitable for adding here.

Reply:
The first part of the PID is now really locked down